org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/guice/AbstractGuiceAwareExecutableExtensionFactory.java

```diff
@@ -49,15 +50,35 @@ public Object create() throws CoreException {
 		try {
 			final Class<?> clazz = getBundle().loadClass(clazzName);
 			final Injector injector = getInjector();
+			if (injector == null) {
+				throw handleCreationError(null);
+			}
 			final Object result = injector.getInstance(clazz);
 			if (result instanceof IExecutableExtension)
 				((IExecutableExtension) result).setInitializationData(config, null, null);
 			return result;
+		} catch (Throwable e) {
```

Comment:
Throwable is intended here. During instantiation of the class also an Error, e.g. NoClassDefFoundError, can occur. Also in this case the UI cannot operate properly and the user gets a better error message. Does this make sense for you?